### ISSUE TITLE
[Abandoned Plugin] TouchUI

### ISSUE THREAD

**Issue by roblandry** [OctoPrint/plugins.octoprint.org#1126]
### Which Plugin?

https://plugins.octoprint.org/plugins/touchui/

### Link to ticket on plugin's issue tracker confirming abandonment

https://github.com/BillyBlaze/OctoPrint-TouchUI/issues/535

**Comment by wwian** [comment]
<sad face. I still use this app with my prints.>

**Comment by jose-campos** [comment]
What’s the process to take this over? Fork and re-submit?

**Comment by jneilliii** [comment]
Fork, address any outstanding major issues relative to compatibility, etc. bump version and submit PR to redirect the update checks. A good example can be found [here](https://github.com/OctoPrint/plugins.octoprint.org/pull/1041/files) of such a PR.

**Comment by jose-campos** [comment]
Sounds good, will install and check the issues to see how much work may be involved and report back.

**Comment by vexofp** [comment]
Probably the biggest issue is an incompatibility with Octoprint 1.8.0+. See https://github.com/BillyBlaze/OctoPrint-TouchUI/issues/532 for more info.

There is a link in there to a branch I made and have been using locally that fixed that problem, but I was never entirely happy with it because it broke backwards compatibility with older Octoprint versions. Maybe that is less of a problem now, since 1.8.0 has been out for awhile now? Either way, hopefully it helps!

**Comment by JohnathonMohr** [comment]
I've been wondering what happened to the temp graph. 😅

Well, if the plugin transfers ownership, that would provide a convenient time to shift compatibility to 1.8.0+. Just my 2 cents.

**Comment by sfarkhan** [comment]
I use TouchUI as well. I would be up to Patreon someone to take over. It's a popular plugin

**Comment by TheNore** [comment]
Has there been any update on this? I use this on all of my printers, I would gladly donate to whoever took this over, its a great Plugin!! It would be ashamed to loose it.

**Comment by jneilliii** [comment]
> but I was never entirely happy with it because it broke backwards compatibility with older Octoprint versions

You could do a copy of the file and conditionally load the different file based on OctoPrint version. Here's an example of where I've used this.

https://github.com/jneilliii/OctoPrint-TPLinkSmartplug/blob/27d265aadf164f2f2aebad99ef64874af029d869/octoprint_tplinksmartplug/__init__.py#L337

Also, it's always better to utilize id css selector if at all possible, just not sure without looking directly at it if that's available here or not.

**Comment by Thierry-Myron** [comment]
So sad... Still using on 10.1 touchscreen :(

**Comment by Unn4m3DD** [comment]
I'm considering taking a look at the plugin, fixing the graph, and maybe some quality-of-life updates, just asking here if someone else is already on it so that we don't duplicate our efforts

**Comment by Unn4m3DD** [comment]
Well, I have it sorted out. I've created a fork with the changes, can someone help me with how to make the update reach octoprints repositories? https://github.com/Unn4m3DD/OctoPrint-TouchUI
![image](https://user-images.githubusercontent.com/42273661/222244359-0d1d0d70-6130-4a56-8e83-9c8402ef8c7d.png)

**Comment by vexofp** [comment]
@Unn4m3DD , I think the legend and temperature values at the bottom there still look messed up. They should look like this:
![image](https://user-images.githubusercontent.com/15070192/222935133-a02ea2c8-5d24-48b0-bd79-b72b35de44fc.png)

Please feel free to steal from https://github.com/BillyBlaze/OctoPrint-TouchUI/compare/master...vexofp:532-temp-graph-1.8.x (if helpful)!

**Comment by Unn4m3DD** [comment]
I thought those bars were sliders xD I managed to get to this, should be close enough
<img width="407" alt="image" src="https://user-images.githubusercontent.com/42273661/223824962-1badea1d-04df-4f8d-9537-6b39e4a7155a.png">

**Comment by Unn4m3DD** [comment]
Heyy, I have had the PR opened for some time and I'm wondering if I did it in the correct way. @vexofp, you seem to know your way around the octoprint community, any chance you might be able to help me understand this?

**Comment by jneilliii** [comment]
If you want to adopt the plugin it's just a matter of submitting a PR here that configures the correct update check overlays, the plugin listing data adding you as an author and a bumped version with release on your fork of the plugin. Here is an example of a completed adoption PR to see what I mean.

https://github.com/OctoPrint/plugins.octoprint.org/pull/1157/files

**Comment by ThatMattCraig** [comment]
I'll take this over. Ill try my best to keep it running for now

**Comment by pzwe2020** [comment]
My friend and i would love to transform this project and fix it. We will work our hardest on keeping it up. We just need someone to convert everything to python! I am new to get hub, and need to know how to take over the project.

**Comment by pzwe2020** [comment]
*Github

**Comment by Mbayazeed** [comment]
can't get a cam stream either !!

**Comment by devildant** [comment]
Good morning,
unfortunately I don't have time to resume this plugin, however I made a fix for the webcam on my fork if you want:
https://github.com/devildant/OctoPrint-TouchUI

just download the master branch as zip, and install manually inside octoprint

**Comment by devildant** [comment]
Note: I applied the fix from @Unn4m3DD + fix temp graph settings + fix the SD card symbol in the files section

**Comment by devildant** [comment]
![image](https://github.com/OctoPrint/plugins.octoprint.org/assets/8700106/d3b28de0-7c39-4837-a684-cf7371fccf69)


![image](https://github.com/OctoPrint/plugins.octoprint.org/assets/8700106/3863f462-839b-4174-9354-feabf0378e85)


![image](https://github.com/OctoPrint/plugins.octoprint.org/assets/8700106/c72f1092-a499-4605-aa41-4572d9e61551)

**Comment by Mbayazeed** [comment]
That worked perfectly.

Thanks

Kindest regards .

Sent from BlueMail<https://bluemail.me>
On Jun 22, 2023, at 9:16 PM, devildant ***@***.******@***.***>> wrote:

Good morning,
unfortunately I don't have time to resume this plugin, however I made a fix for the webcam on my fork if you want:
https://github.com/devildant/OctoPrint-TouchUI

just download the master branch as zip, and install manually inside octoprint

**Comment by chuwy5** [comment]
is someone maintaining it? if not depending on the code I might be willing to tip my hat in to atempt to keep it going

**Comment by devildant** [comment]
> is someone maintaining it? if not depending on the code I might be willing to tip my hat in to atempt to keep it going

at the moment there is no one to take over, i just made patches on my branch to fix the problems on the latest version of octoprint. if you want to resume the project, go go :), it will please a lot of people

**Comment by pzwe2020** [comment]
I have been working on trying to figure out the code and I just can get it. Hopefully we can find someone else. Sorry

On August 5, 2023, at 12:33 PM, chuwy5 ***@***.***> wrote:

is someone maintaining it? if not depending on the code I might be willing to tip my hat in to atempt to keep it going

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.￼Message ID: ***@***.***>

[ { ***@***.***": "<a href="http://schema.org">http://schema.org</a>", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "<a href="https://github.com/OctoPrint/plugins.octoprint.org/issues/1126#issuecomment-1666587788">https://github.com/OctoPrint/plugins.octoprint.org/issues/1126#issuecomment-1666587788</a>", "url": "<a href="https://github.com/OctoPrint/plugins.octoprint.org/issues/1126#issuecomment-1666587788">https://github.com/OctoPrint/plugins.octoprint.org/issues/1126#issuecomment-1666587788</a>", "name": "View Issue" }, "description": "View this Issue on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "<a href="https://github.com">https://github.com</a>" } } ]

**Comment by chuwy5** [comment]
no worries. my thought is to find a group of people that can work on it.
unless i can figure it out and start maintaining it. but im down to work on
it with others

**Comment by JayroJones** [comment]
Jesus fucking christ, the plugin STILL works fine, and I'm tired of the UI nagging me to remove it because it's "abandoned" by it's creator. (Thanks for that, _douche_.) How do I get it to stop nagging my system, even-though I always hit "Mark as read."? The nag is annoying as fuck to deal with, but the plugin is too useful to get rid of.

**Comment by cp2004** [comment]
@JayroJones - I suggest you choose a different tone so that you come across as a reasonable human being. Please check https://octoprint.org/conduct/.

Marking as read sets a cookie so the notification doesn't show again. If you're having trouble with this, I'd suggest politely trying to seek help on the community forums with appropriate gratitude for anyone who now decides to help you, and remember first impressions count.

**Comment by K1Printing** [comment]
If someone adopts this plugin and continues it's maintenance I'd be happy to chip in via patreon or something like that, I would take a crack at it, but I don't have the time or resources to learn programming at this given moment.

**Comment by heinzight** [comment]
Help please...  I installed this but now I don't want it and there is no 'wrench' to get to be able to uninstall it. How do I uninstall it, please.

**Comment by jneilliii** [comment]
Try http://octopi.local/recovery then restart in safe mode and then you should be able to remove the plugin.

**Comment by heinzight** [comment]
Yes, that worked although I wasn't sure what or where to do the recovery. So it makes me wonder why this plug-in was not on the Blacklist if it's not maintained or at least the warning before not after install.

It seems that install/uninstall wiped out my login credentials (??) to the Pi 4B. I cannot get into the Pi from my computer. I took very careful notes in installing Octopi but it will not acept my username nor password. I can no longer see my files. I can't get into Octopi.org either (that one I do not have notes, long ago when I had a Pi3B.

**Comment by heinzight** [comment]
Thank you Gina!

**Comment by sophia-crypto** [comment]
i am willing to adopt it

**Comment by foosel** [comment]
Yeah... no. Sorry to say, but that just smells wrong with that name suffix and that empty an account.

**Comment by ender-bit** [comment]
I will like to take ownership of the plugin, please let me know if you're interested in giving me access to the plugin

**Comment by rpi10** [comment]
> ### Which Plugin?
> https://plugins.octoprint.org/plugins/touchui/
> 
> ### Link to ticket on plugin's issue tracker confirming abandonment
> [BillyBlaze/OctoPrint-TouchUI#535](https://github.com/BillyBlaze/OctoPrint-TouchUI/issues/535)
> 
> ### Is there someone open to adopting it?
> _No response_

I would like to take back the project. Your work has been truly amazing: THANK YOU!

**Comment by ender-bit** [comment]
To touch ui

**Comment by ender-bit** [comment]
To Touch UI, thats the plugin I would like to take ownership of

**Comment by DanVerseDev** [comment]
🙌 Someone is making a progress in this? i am willing to help mantain this project alive as it's the best of this kind

**Comment by cameleonnzl** [comment]
So the last push on this was a little over four years ago. Do people really like it enough to want it kept alive? I had it in my download list and it's warned as abandoned.

**Comment by TheNore** [comment]
Thats concerning :( I have been using this plugin for years, and would really like to see it updated

**Comment by vexofp** [comment]
> So the last push on this was a little over four years ago. Do people really like it enough to want it kept alive? I had it in my download list and it's warned as abandoned.

Are there other alternatives for a good interface for small screens and mobile devices?

Touch UI has been my daily driver for close to a decade now. Despite the lack of updates, it has luckily managed to escape major breakage over the years.

**Comment by jneilliii** [comment]
OctoDash, which currently is an external app, is currently being refactored into a plugin with kiosk browser mode (the way touchui works on pi touch screen). Just because a plugin is abandoned didn't mean it's going to get deleted and you can continue using it until it breaks.
